sensor/kubernetes/eventpipeline/pipeline_impl.go

```diff
@@ -0,0 +1,72 @@
+package eventpipeline
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type eventPipeline struct {
+	output     component.OutputQueue
+	components []component.PipelineComponent
```

Comment:
I don't see the advantage of having `components` slice. The only benefit is being able to iterate on them on `Stop()`. But since there won't be many, listing them and calling directly seems fine to me. IMO it makes it less clear to me what components are there, because I saw both a slice of `component.PipelineComponent` and a `component.OutputQueue` one. 

I'll leave it up to you if you want to keep it this way. I don't have strong arguments against it.

Reply:
That was probably over architecting on my side. Once the components are created, the `eventPipeline` only needs to know about the output to read it; therefore, we can create the PipelineComponent abstraction and just iterate through it when we want to start or stop. But you are right, there are only three components so I think I’ll just change it to call the components directly.